app/routes/api.js

```diff
@@ -169,4 +170,92 @@ router.post('/report', (req, res) => {
     res.redirect('back');
 });
 
+// API endpoint to delete an attachment for a given attachment ID
+router.delete('/attachment/:attachmentID', (req, res, next) => {
+    res.setHeader('Content-Type', 'application/json')
+    try {
+        const attachmentID = req.params.attachmentID;
+
+        const result = deleteAttachmentforFactoid(attachmentID)
+
+        if (result) {
+            return res.status(200).send({ message: 'Attachment deleted successfully.' });
+        } else {
+            return res.status(404).send({ message: 'Attachment not found.' });
+        }
+    } catch (e) {
+        console.log(e);
+        return res.status(500).send({ message: 'Server error.' });
+    }
+});
+
+// API endpoint to delete a tag for a given factoid ID and category ID
+router.delete('/tag/:factoidID/:categoryID', (req, res) => {
+    res.setHeader('Content-Type', 'application/json')
+    try {
+        const factoidID = req.params.factoidID;
+        const categoryID = req.params.categoryID;
+
+        const result = deleteTagforFactoid(parseInt(factoidID), parseInt(categoryID));
+
+        if (result) {
+            return res.status(200).send({ message: 'Tag deleted successfully.' });
+        } else {
+            return res.status(404).send({ message: 'Tag not found.' });
+        }
+    } catch (e) {
+        console.log(e);
+        return res.status(500).send({ message: 'Server error.' });
+    }
+});
+
+// API endpoint to delete a fact and associated tags and attachments
+router.delete('/fact/:factoidID', (req, res) => {
+    res.setHeader('Content-Type', 'application/json')
+    try {
+        const factoidID = req.params.factoidID;
+        
+        // Delete all attachments associated with the factoid
+        const attachmentsDeleted = deleteAllAttachmentsforFactoid(factoidID);
+        if (!attachmentsDeleted) {
+            return res.status(500).send({ message: 'Error deleting attachments for the factoid.' });
+        }
+
+        // Delete all tags associated with the factoid
+        const tagsDeleted = deleteAllTagsforFactoid(factoidID);
+        if (!tagsDeleted) {
+            return res.status(500).send({ message: 'Error deleting tags for the factoid.' });
+        }
+
+        // Delete the factoid itself
+        const factDeleted = deleteFactByID(factoidID);
+        if (factDeleted) {
+            return res.status(200).send({ message: 'Fact and associated attachments/tags deleted successfully.' });
+        } else {
+            return res.status(404).send({ message: 'Error deleting Factoid.' });
+        }
+    } catch (e) {
+        console.log(e);
+        return res.status(500).send({ message: 'Server error.' });
+    }
+});
+
+// API endpoint to approve a fact with the given id.
+router.put('/approve/:factoidID', (req, res, next) => {
+    res.setHeader('Content-Type', 'application/json')
+    try {
+        let factoidID = req.params.factoidID
+
```

Comment:
Pretty much same as above :)
```suggestion
        const factoidID = parseInt(req.params.factoidID);
        
		if (isNaN(factoidID)) {
			return res.status(404).send({ message: 'Invalid factoid ID in request.' });
		}
```